### DISCUSSION TITLE
upgrade v1.2205.0 to v1.2206.0

### DISCUSSION THREAD

**Discussion by sjjian** [actiontech/sqle#643] [Upgrade]
## Feature
[#565] 支持 TiDB 类型数据库审核【企业版】
[#573] 新增5条TiDB专属规则【企业版】
1. 禁用 non-full group by 语法
2. group_concat() 语法必须使用order by
3. limit 分页查询必须使用排序
4. 未使用自增主键/自动随机主键写入
5. 高并发写入表不建议使用自增主键

[#568] SQL查询支持查看数据字典和执行计划，审核SQL支持进行SQL分析【企业版】；
[#589] SQLE license内容调整，支持根据数据库类型进行细分限制【企业版】；
[#606] 登录页面样式调整；

## Improvement
[#633] 工单的审核通过率百分比仅展示小数点后2位;
[#628] 优化新增审核规则模版或者是克隆审核规则模版都很耗时的问题。

## Bugfix
[#563] 修复SQLE scanner 解析带有空otherwise标签的 mybatis XML时异常崩溃（社区反馈）
[#597] 修复在SQLE使用MySQL8.0作为业务库时，通过open api往审核任务中添加SQL时，接口报错报错的问题（社区反馈);
[#594] 修复使用数据源创建过工单之后，对应的删除数据源若删除会导致工单详情页面打开提示未知错误的问题；
[#641] 修复审核时如果当前连接的表存在但是表格是个空表，审核失败的问题；
[#634] 修复SQL查询时执行计划tab页无法关闭的问题；
[#624] 修复未设置运维时间时，无法选择所有的时间进行定时上线的问题；
[#621] 修复Oauth2账户绑定成功后，再次登录依旧跳转到用用户绑定界面的问题；
[#597] 修复当SQLE后端存储未MySQL8.0版本时，通过open api往审核任务中添加sql时，部分数据报错的问题；
[#639] 修复对特定语法（choose内不带otherwise标签）的Mybatis 文件审核时程序panic的问题。

<hr /><em>This discussion was created from the release <a href='https://github.com/actiontech/sqle/releases/tag/v1.2206.0'>v1.2206.0</a>.</em>

**Comment by ColdWaterLW** [comment]
### 老版本升级
新部署的sqle audit_plan_report_sqls_v2表增加number字段，修改方式:
1. 登入SQLE存储元数据的数据库
2. 执行如下存储过程，以下SQL假设SQLE业务库为`sqle`
```sql
DELIMITER $$
CREATE PROCEDURE updateAuditPlanReport()
BEGIN
  DECLARE report_end_flag int DEFAULT 0;
  DECLARE last_report_id int DEFAULT 0;
  DECLARE num int DEFAULT 1;
  DECLARE val_id int(10) unsigned;
  DECLARE val_audit_plan_report_id int(10) unsigned;
  DECLARE val_number int(10) unsigned;
  DECLARE cur_report CURSOR FOR SELECT id,audit_plan_report_id,number FROM sqle.audit_plan_report_sqls_v2 WHERE number IS NULL AND deleted_at IS NULL;
  DECLARE CONTINUE HANDLER FOR NOT FOUND SET report_end_flag=1;
  
  OPEN cur_report;
  FETCH cur_report INTO val_id,val_audit_plan_report_id,val_number;
  WHILE report_end_flag <>1 DO
    IF val_audit_plan_report_id!=last_report_id THEN
      SET num=1;
      SET last_report_id=val_audit_plan_report_id;
    END IF;
    UPDATE sqle.audit_plan_report_sqls_v2
    SET number=num
    WHERE id=val_id;
    SET num=num+1;
    FETCH cur_report INTO val_id,val_audit_plan_report_id,val_number;
  END WHILE;
END; 
$$
DELIMITER ;
CALL updateAuditPlanReport();
```
